ibis/backends/duckdb/__init__.py

```diff
@@ -534,6 +544,25 @@ def attach_sqlite(
                 sa.text(f"CALL sqlite_attach('{str(path)}', overwrite={overwrite})")
             )
 
+    def _register_in_memory_tables(self, expr: ir.Expr) -> None:
+        from ibis.expr.analysis import find_physical_tables
+
+        # Warn for any tables depending on RecordBatchReaders that have already
+        # started being consumed.
+        for t in find_physical_tables(expr.op()):
+            started = self._record_batch_reader_states.get(t.name)
+            if started is True:
+                warnings.warn(
+                    f"Table {t.name!r} is backed by a `pyarrow.RecordBatchReader` "
+                    "that has already been partially consumed. This may lead to "
+                    "incorrect results. Either recreate the table from a new "
+                    "`pyarrow.RecordBatchReader`, or use `con.create_table` to "
```

Comment:
Perhaps we should recommend `con.create_table` or `.cache()` here as the "obvious" thing to do to persist the thing (one permanent, one temporary but not ephemeral).